saleor/graphql/channel/dataloaders.py

```diff
@@ -113,3 +115,33 @@ def map_shipping_zones(shipping_zones):
             .load_many({pk for pk, _ in zone_and_channel_is_pairs})
             .then(map_shipping_zones)
         )
+
+
+class WarehousesByChannelIdLoader(DataLoader):
+    context_key = "warehouse_by_channel"
+
+    def batch_load(self, keys):
+        warehouse_and_channel_in_pairs = (
+            Warehouse.objects.using(self.database_connection_name)
+            .filter(channels__id__in=keys)
+            .values_list("pk", "channels__id")
+        )
```

Comment:
This soultion should be better for database. 
```suggestion
        WarehouseChannel = Warehouse.channels.through
        warehouse_and_channel_in_pairs = (
            WarehouseChannel.objects.using(self.database_connection_name)
            .filter(channel_id__in=keys)
            .values_list("warehouse_id", "channel_id")
        )
```

Reply:
Good point. Thank you!